plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -901,6 +929,55 @@ public void testTextualPredicatePushdown()
         assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name = 'romania'"))
                 .returnsEmptyResult()
                 .isFullyPushedDown();
+        assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name = 'romania' OR comment = 'P'"))
+                .returnsEmptyResult()
+                .isFullyPushedDown();
+
+        String withConnectorExpression = " OR some_column = 'x'";
+        try (TestTable table = new TestTable(
+                onRemoteDatabase(),
+                "tpch.test_textual_predicate_pushdown",
+                """
+                        (
+                        some_column String,
+                        a_string String,
+                        a_string_alias text,
+                        a_fixed_string FixedString(1),
+                        a_string_nullable Nullable(String),
+                        a_string_alias_nullable Nullable(text),
+                        a_fixed_string_nullable Nullable(FixedString(1)),
+                        a_enum_1 Enum('hello', 'world', 'a', 'b', 'c'),
+                        a_enum_2 Enum('hello', 'world', 'a', 'b', 'c'))
```

Comment:
Can we move this to a dedicated test for enums ?

Reply:
It was initially implemented as a dedicated test, however it is duplicated in this test anyway to support:
```
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string = a_string_alias")).isFullyPushedDown();
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string = a_string_alias" + withConnectorExpression)).isFullyPushedDown();
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string = a_enum_1")).isNotFullyPushedDown(FilterNode.class);
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string = a_enum_1" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
``` 
scenarios. So let it be as it is.